MoEngagePluginBase/MoEngagePlugin.swift

```diff
@@ -53,32 +59,39 @@ import MoEngageInApps
         initializeMoEngageSecondaryInstance(sdkConfig: sdkConfig, launchOptions: launchOptions)
     }
     
+    private func initializeMoEngageSecondaryInstance(sdkConfig: MOSDKConfig, sdkState: MoEngageSDKState) {
+       #if DEBUG
+         MoEngage.sharedInstance().initializeTestInstance(with: sdkConfig, andSDKState: sdkState)
+      #else
+        MoEngage.sharedInstance().initializeLiveInstance(with: sdkConfig, andSDKState: sdkState)
+      #endif
+        
+     commonSetUp(identifier: sdkConfig.identifier)
+
+    }
+    
     private func initializeMoEngageSecondaryInstance(sdkConfig: MOSDKConfig, launchOptions: [UIApplication.LaunchOptionsKey: Any]? = nil) {
         #if DEBUG
             MoEngage.sharedInstance().initializeTestInstance(with: sdkConfig, andLaunchOptions: launchOptions)
         #else
             MoEngage.sharedInstance().initializeLiveInstance(with: sdkConfig, andLaunchOptions: launchOptions)
         #endif
         
-        setDelegates(identifier: sdkConfig.identifier)
+        commonSetUp(identifier: sdkConfig.identifier)
     }
     
     @objc public func trackPluginInfo(_ pluginType: String, version: String) {
         let integrationInfo = MOIntegrationInfo(pluginType: pluginType, version: version)
         MOCoreIntegrator.sharedInstance.addIntergrationInfo(info: integrationInfo)
     }
     
+    private func commonSetUp(identifier: String) {
```

Comment:
@RakshithaAcharya Was there any reason for creating another function for setDelegates here?

Reply:
@deepa-iOS-SDK Moving forward if there is any code that should be called post initialization(with different initialization methods), will go inside crommonsetup.